### ISSUE TITLE
Extend kubectl config to support ComponentConfigs?

### ISSUE THREAD

**Issue by abiogenesis-now** [kubernetes/kubernetes#50379]
/kind feature

Original suggestion from @brugz, based on discussion in https://github.com/kubernetes/kubernetes/issues/50041#issuecomment-321160158. 

Since the ComponentConfig object for things like `kube-proxy` are pulled from a config file, seemed like it would be nice to have a `kubectl` to make it easy to edit said file. This could basically be `kubectl config` but maybe with a `--kind` flag or something to specify which component it's for.

Not sure if this makes sense design-wise/implementation-wise, but thought it was worth bringing up.

(*Tangential question:* if I edit a ComponentConfig directly through `kubectl`, will it change the config file? Or will it be overwritten by the config file settings again after `config-sync-period`?)

**Comment by ncdc** [comment]
I don't think this is exactly what was suggested in the other issue. The request was to be able to use `kubectl` to view a component's active configuration.

**Comment by abiogenesis-now** [comment]
@ncdc Updated description, let me know if that clears things up (or if further edits should be made).
